soap/client.go

```diff
@@ -81,6 +83,10 @@ func (c *Client) RoundTrip(in, out Message) error {
 		return err
 	}
 	defer resp.Body.Close()
+	if resp.StatusCode != http.StatusOK {
+		body, _ := ioutil.ReadAll(resp.Body)
```

Comment:
Can you wrap `resp.Body` into an `io.LimitedReader` and only read the first few MBs of the response at most?